supermarq/feature_plotting.py

```diff
@@ -0,0 +1,132 @@
+import matplotlib.pyplot as plt
+import numpy as np
+from matplotlib.patches import Circle, RegularPolygon
+from matplotlib.path import Path
+from matplotlib.projections import register_projection
+from matplotlib.projections.polar import PolarAxes
+from matplotlib.spines import Spine
+from matplotlib.transforms import Affine2D
+
+
+def radar_factory(num_vars, frame="circle"):
+    """
+    (https://matplotlib.org/stable/gallery/specialty_plots/radar_chart.html)
+
+    Create a radar chart with `num_vars` axes.
+
+    This function creates a RadarAxes projection and registers it.
+
+    Parameters
+    ----------
+    num_vars : int
+        Number of variables for radar chart.
+    frame : {'circle', 'polygon'}
+        Shape of frame surrounding axes.
+
+    """
+    # calculate evenly-spaced axis angles
+    theta = np.linspace(0, 2 * np.pi, num_vars, endpoint=False)
+
+    class RadarAxes(PolarAxes):
```

Comment:
Move this class outside this function.

Reply:
Is it possible to keep `RadarAxes` inside of `radar_factory`? I think this is how these matplotlib factories are typically defined. This is because `RadarAxes` needs to know the value of `num_vars`, `frame`, and `theta`, but it is only instantiated on line 29: 

`fig, ax = plt.subplots(dpi=150, subplot_kw=dict(projection="radar"))`

So defining the class within `radar_factory` gives it access to those local variables

Reply:
I can discuss this more if you'd like to hop on a call quick. But I just tried moving it outside and passing those variables to the constructor but this caused other problems with the variables that matplotlib is passing to `RadarAxes` under the hood

Reply:
What we discussed: Create 

```
class RadarAxesMeta(PolarAxes):
```
and 
```
def radar_factory(num_vars, frame="circle"):
      class RadarAxes(RadaAxesMeta):
```